### PR TITLE
Minor context traversal additions + tutorial section

### PR DESCRIPTION
**Pull request by ajratner**
Minor edits:
* Add `get_parent()` and `get_children()` methods to `Context`.  Having these be methods is both simple and allows for custom specification of e.g. how the children are sorted
* Adds a small section to tutorial 2

### CODE REVIEW

File: snorkel/models/context.py [hunk 1]
```diff
@@ -296,7 +314,13 @@ class Span(Context, TemporarySpan):
         'inherit_condition': (id == Context.id)
     }
 
-    parent = relationship('Context', backref=backref('spans', cascade='all, delete-orphan'), foreign_keys=parent_id)
+    parent = relationship('Context', backref=backref('spans', cascade='all, delete-orphan'), order_by=char_start, foreign_keys=parent_id)
```
Comment (henryre) [inline snorkel/models/context.py]: Should we name this sentence for consistency with `Sentence.document`?

File: tutorials/intro/Intro_Tutorial_2.ipynb [hunk 2]
```diff
@@ -279,8 +279,70 @@
...
+   "source": [
+    "c = train_cands[0]\n",
+    "c.get_contexts()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "These `Context` objects are part of a hierarchy of `Context` objects in Snorkel.  In our case, this hierarchy consists of `Document`s, `Sentence`s, and `Span`s.  We can traverse this hierarchy by using the specific names--e.g., `doc.sentences`--or by using the generic `get_parent()` and `get_children()` methods:"
```
Comment (henryre) [inline tutorials/intro/Intro_Tutorial_2.ipynb]: This should probably be merged with and come before the explanation of `Candidate`s being composed of `Context`s

File: snorkel/models/context.py (hunk 1 above)
Comment (ajratner) [inline snorkel/models/context.py]: Yup good call

File: tutorials/intro/Intro_Tutorial_2.ipynb (hunk 2 above)
Comment (ajratner) [inline tutorials/intro/Intro_Tutorial_2.ipynb]: Yup, thanks!  Making edits now

**Pull request merged**